src/Message/PurchaseRequest.php

```diff
@@ -73,6 +54,40 @@ public function getData()
 
     public function sendData($data)
     {
-        return $this->response = new PurchaseResponse($this, $data);
+        return $this->response = new PaymentRedirectResponse($this, $data);
+    }
+
+    public function setCreateCard($value)
+    {
+        return $this->setParameter('createCard', $value);
+    }
+
+    public function isCreatingCard()
+    {
+        return true === $this->getParameter('createCard', false);
+    }
+
+    public function setCardReference($value)
+    {
+        return $this->setParameter('cardReference', $value);
+    }
+
+    public function getCardReference()
+    {
+        return $this->getParameter('cardReference');
+    }
+
+    public function hasCardReference()
+    {
+        return null !== $this->getCardReference();
+    }
+
+    private function guessPageAction()
```

Comment:
`definePageAction()` or `resolvePageAction()` ... :thinking:

Reply:
`getPageAction()` ?

Reply:
`resolvePageAction` seems ok to me

Reply:
I was afraid that we mistake this with a simple getter, hence define or resolve. What do you think?

Reply:
I think there is no simple getter. But resolve seems good. Go for it